p2p/transport/webrtc/udpmux/mux.go

```diff
@@ -0,0 +1,278 @@
+package udpmux
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net"
+	"sync"
+
+	logging "github.com/ipfs/go-log/v2"
+	pool "github.com/libp2p/go-buffer-pool"
+	"github.com/pion/ice/v2"
+	"github.com/pion/stun"
+)
+
+var log = logging.Logger("webrtc-udpmux")
+
+const ReceiveMTU = 1500
+
+// UDPMux multiplexes multiple ICE connections over a single net.PacketConn,
+// generally a UDP socket.
+//
+// The connections are indexed by (ufrag, IP address family)
+// and by remote address from which the connection has received valid STUN/RTC
+// packets.
+//
+// When a new packet is received on the underlying net.PacketConn, we
+// first check the address map to see if there is a connection associated with the
+// remote address. If found we forward the packet to the connection. If an associated
+// connection is not found, we check to see if the packet is a STUN packet. We then
+// fetch the ufrag of the remote from the STUN packet and use it to check if there
+// is a connection associated with the (ufrag, IP address family) pair. If found
+// we add the association to the address map. If not found, it is a previously
+// unseen IP address and the `unknownUfragCallback` callback is invoked.
+type UDPMux struct {
+	socket               net.PacketConn
+	unknownUfragCallback func(string, net.Addr) bool
```

Comment:
You're right, this is (again) just terrible API design. Following Filippo's rule "You'll never regret returning an error instead of a bool", I think it's much cleaner to just make this an error. No documentation needed.

Fixed it in 
89ad9385ca88e23ac006caebf2317d6a0075f96f.